packages/nextjs-mf/src/plugins/NextFederationPlugin.ts

```diff
@@ -38,187 +51,473 @@ const regexEqual = (x, y) => {
+          `[nextjs-mf] You are sharing ${key} from the default share scope. This is not necessary and can be removed.`
+        );
+        // Use a type assertion to inform TypeScript that 'key' can be used as an index for the 'shared' object
+        delete (shared as { [key: string]: unknown })[key];
+      }
+      return acc;
+    },
+    []
+  );
+
+  if (warnings.length > 0) {
+    console.warn('%c' + warnings.join('\n'), 'color: red');
+  }
+}
+
+/**
+ * Applies client-specific plugins.
+ * @param {Compiler} compiler - The Webpack compiler instance.
+ * @param options
+ * @param extraOptions
+ */
+function applyClientPlugins(
+  compiler: Compiler,
+  options: ModuleFederationPluginOptions,
+  extraOptions: NextFederationPluginExtraOptions
+): void {
+  const { webpack } = compiler;
+  const { remotes, name } = options;
+
+  new AddModulesPlugin({
+    runtime: name,
+    eager: false,
+    remotes,
+  }).apply(compiler);
+
+  new AddModulesPlugin({
+    runtime: 'webpack',
+    eager: true,
+    remotes,
+    shared: DEFAULT_SHARE_SCOPE_BROWSER,
+    container: name + '_single',
+  }).apply(compiler);
+
+  if (extraOptions.automaticPageStitching) {
+    compiler.options.module.rules.push({
+      test: /next[\\/]dist[\\/]client[\\/]page-loader\.js$/,
+      loader: path.resolve(__dirname, '../loaders/patchNextClientPageLoader'),
+    });
+  }
+
+  if (options.library) {
+    console.error('[nextjs-mf] you cannot set custom library');
+  }
+
+  options.library = {
+    type: 'window',
+    name,
+  };
+
+  new webpack.EntryPlugin(
+    compiler.context,
+    require.resolve('../internal-delegate-hoist'),
+    'main'
+  ).apply(compiler);
+
+  new ChunkCorrelationPlugin({
+    filename: 'static/chunks/federated-stats.json',
+  }).apply(compiler);
+
+  new CommonJsChunkLoadingPlugin({
+    asyncChunkLoading: true,
+    name: options.name,
+    remotes: options.remotes as Record<string, string>,
+    baseURI: compiler.options.output.publicPath,
+    verbose: false,
+  }).apply(compiler);
+}
+
+/**
+ * Gets the appropriate ModuleFederationPlugin based on the environment.
+ * @param {boolean} isServer - A flag to indicate if the environment is server-side or not.
+ * @param {Compiler} compiler - The Webpack compiler instance.
+ * @returns {ModuleFederationPlugin | undefined} The ModuleFederationPlugin or undefined if not applicable.
+ */
+function getModuleFederationPluginConstructor(
+  isServer: boolean,
+  compiler: Compiler
+): ConstructableModuleFederationPlugin {
+  if (isServer) {
+    return require('@module-federation/node')
+      .NodeFederationPlugin as ConstructableModuleFederationPlugin;
+  } else {
+    return compiler.webpack.container
+      .ModuleFederationPlugin as unknown as ConstructableModuleFederationPlugin;
+  }
+}
+
+/**
+ * Set up default shared values based on the environment.
+ * @param isServer - Boolean indicating if the code is running on the server.
+ */
+const setDefaultShared = (isServer: boolean) => {
```

Comment:
This function is not setting anything, maybe this should be called `retrieveDefaultShared`?